evm/estimators/gas.py

```diff
@@ -25,8 +37,52 @@ def execute_plus_buffer(multiplier, state, transaction):
 double_execution_cost = execute_plus_buffer(2)
 
 
+def spoof_execute_transaction(state, transaction):
+
+    if isinstance(transaction, SpoofTransaction):
+        _transaction = transaction
+    else:
+        _transaction = SpoofTransaction(transaction)
+
+    setattr(_transaction, 'v', DEFAULT_SPOOF_V)
+    setattr(_transaction, 's', DEFAULT_SPOOF_S)
+    setattr(_transaction, 'r', DEFAULT_SPOOF_R)
+
+
+    snapshot = state.snapshot()
+    try:
+        with state.mutable_state_db() as state_db:
+            if not hasattr(_transaction, 'get_sender'):
+                setattr(
+                    _transaction,
+                    'get_sender',
+                    lambda: to_bytes(hexstr=DEFAULT_SPOOF_SENDER))
+
+                setattr(
+                    _transaction,
+                    'sender',
+                    to_bytes(hexstr=DEFAULT_SPOOF_SENDER))
+
+                setattr(
+                    _transaction,
+                    'nonce',
+                    DEFAULT_SPOOF_NONCE)
+
+            # set the account balance of the sender to an arbitrary large
+            # amount to ensure they have the necessary funds to pay for the
+            # transaction.
+            state_db.set_balance(transaction.sender, UINT_256_MAX // 2)
```

Comment:
Also, I think this might hide failures we care about. Maybe we could spoof gas_price to 0 instead? If the transaction is trying to send funds in `value`, and the account doesn't have enough funds, I think that estimate gas shouldn't pretend that it succeeds.

Reply:
Thats a good idea.  Pre-funding is what geth does, but just making the actual cost 0 seems better.